drivers/npmx/shell/ldsw.c

```diff
@@ -115,32 +129,43 @@ static int cmd_ldsw_gpio_set(const struct shell *shell, size_t argc, char **argv
 		return 0;
 	}
 
-	int gpio_index = args_info.arg[1].result.ivalue;
-	npmx_ldsw_gpio_config_t gpio_config = {
-		.gpio = ldsw_gpio_index_convert(gpio_index),
-		.inverted = args_info.arg[2].result.bvalue,
-	};
-
-	if (gpio_config.gpio == NPMX_LDSW_GPIO_INVALID) {
-		shell_error(shell, "Error: wrong GPIO index.");
+	npmx_ldsw_gpio_config_t gpio_config;
+	npmx_error_t err_code = npmx_ldsw_enable_gpio_get(ldsw_instance, &gpio_config);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "GPIO config");
 		return 0;
 	}
 
-	if (!check_pin_configuration_correctness(shell, gpio_index)) {
-		return 0;
+	if (param == LDSW_GPIO_PARAM_INDEX) {
+		gpio_config.gpio = ldsw_gpio_index_convert(args_info.arg[1].result.ivalue);
+
+		if (gpio_config.gpio == NPMX_LDSW_GPIO_INVALID) {
+			shell_error(shell, "Error: wrong GPIO index.");
+			return 0;
+		}
+
+		if (!check_pin_configuration_correctness(shell, gpio_config.gpio)) {
+			return 0;
+		}
+	} else if (param == LDSW_GPIO_PARAM_POLARITY) {
+		gpio_config.inverted = args_info.arg[1].result.bvalue;
 	}
 
-	npmx_error_t err_code = npmx_ldsw_enable_gpio_set(ldsw_instance, &gpio_config);
+	err_code = npmx_ldsw_enable_gpio_set(ldsw_instance, &gpio_config);
 	if (!check_error_code(shell, err_code)) {
 		print_set_error(shell, "GPIO config");
 		return 0;
 	}
 
-	shell_print(shell, "Success: %d %d.", gpio_index, gpio_config.inverted);
+	int16_t *p_value = (param == LDSW_GPIO_PARAM_INDEX) ? (int16_t *)&gpio_config.gpio :
+							      (int16_t *)&gpio_config.inverted;
+
+	shell_print(shell, "Success: %d.", *p_value);
```

Comment:
Should be handled with a `switch-case`.

[example](https://github.com/NordicSemiconductor/npmx-zephyr/blob/develop/drivers/npmx/npmx_shell.c#L2676)

Reply:
There's no logic here, why is `switch` needed?

Reply:
RIght.